.github/workflows/pint-backend.yml

```
@@ -0,0 +1,29 @@
+name: Laravel Pint
+
+on:
+  push:
+    paths:
+      - 'laravel/**'
+  pull_request:
+    paths:
+      - 'laravel/**'
```

指定しているディレクトリが `laravel/**` なので動いていないみたいですね。
以下のようにしてみてもう一度試してみてください。
```suggestion
on:
  push:
    paths:
      - 'backend/**'
  pull_request:
    paths:
      - 'backend/**'
```